grabbit/src/test/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessorSpec.groovy

```diff
@@ -191,4 +195,151 @@ class JcrNodesProcessorSpec extends Specification {
+        NodeProtos.Node nodeProto = jcrNodesProcessor.process(node)
+
+        then:
+        nodeProto != null
+        nodeProto.properties.propertyList.first().value.stringValue == JcrConstants.NT_FILE
+        nodeProto.mandatoryChildNodeList.size() == 1
+        nodeProto.mandatoryChildNodeList.first().name == "testMandatoryChild" //making sure child was processed
+        nodeProto.mandatoryChildNodeList.first().properties.propertyList.first().name == JcrConstants.JCR_PRIMARYTYPE
+        nodeProto.mandatoryChildNodeList.first().properties.propertyList.first().value.stringValue == JcrConstants.NT_UNSTRUCTURED
+    }
+
+
+    @Shared
+    DateTime oldDate = new DateTime(2015, 8, 4, 15, 24, 34, 961)
+
+    @Unroll
+    def "Process a node with old date properties: #propList"() {
+        //tests for all situations with non-deltaContent nodes (not updated since last sync)
+        when:
+        JcrNodesProcessor jcrNodesProcessor = new JcrNodesProcessor()
+        jcrNodesProcessor.setContentAfterDate(oldDate.plusDays(1).toString())
+        NodeProtos.Node nodeProto = jcrNodesProcessor.process(aJcrNode)
+        aJcrNode.properties.toList()
+
+        then:
+        nodeProto == null //not copying old data
+
+        where:
+        aJcrNode << [
+                node("testComponent",
+                        property(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED),
+                        property(JcrConstants.JCR_CREATED, oldDate.toCalendar(Locale.default)),
+                ).build(),
+                node("testComponent",
+                        property(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED),
+                        property(JcrConstants.JCR_LASTMODIFIED, oldDate.toCalendar(Locale.default))
+                ).build(),
+                node("testComponent",
+                        property(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED),
+                        property("cq:lastModified", oldDate.toCalendar(Locale.default))
+                ).build(),
+                node("testComponent",
+                        property(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED),
+                        property(JcrConstants.JCR_CREATED, oldDate.toCalendar(Locale.default)),
+                        property(JcrConstants.JCR_LASTMODIFIED, oldDate.toCalendar(Locale.default))
+                ).build(),
+                node("testComponent",
+                        property(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED),
+                        property(JcrConstants.JCR_CREATED, oldDate.toCalendar(Locale.default)),
+                        property("cq:lastModified", oldDate.toCalendar(Locale.default))
+                ).build(),
+        ]
+        propList = aJcrNode.properties.toList().findAll {it.name != JcrConstants.JCR_PRIMARYTYPE}.collectEntries { [(it.name): new DateTime(it.value.date.time.time)]}
+    }
+
+    @Shared
+    DateTime currentDate = new DateTime()
```

Comment:
Does this need to be shared? This seems to be used only in the test below, the `currentDate` in the 1st test looks like a local variable.